src/code/z_message_nes.c

```diff
@@ -1,18 +1,273 @@
+    o += FONT_CHAR_TEX_SIZE;
+    msgCtx->decodedBuffer[p] = 0x65;
+    p++;
+    Font_LoadCharNES(globalCtx, 0x65U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    msgCtx->decodedBuffer[p] = 0x65;
+    p++;
+    Font_LoadCharNES(globalCtx, 0x73U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    msgCtx->decodedBuffer[p] = 0x73;
+
+    f += 16.0f * msgCtx->unk12098 * 6.0f;
+    *decodedBufPos = p;
+    *offset = o;
+    *arg3 = f;
+}
+
+void Message_LoadLocalizedRupeesNES(GlobalContext* globalCtx, s16* decodedBufPos, s32* offset, f32* arg3) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    s16 p = *decodedBufPos;
+    s32 o = *offset;
+    f32 f = *arg3;
+    u8 j;
+
+    msgCtx->decodedBuffer[p] = 0x20;
+    p = p + 1;
+    for (j = 0; j < D_801D0710[gSaveContext.language - 1]; j++) {
+        Font_LoadCharNES(globalCtx, D_801D06F0[gSaveContext.language - 1][j], o);
+        msgCtx->decodedBuffer[p] = D_801D06F0[gSaveContext.language - 1][j];
+        o += FONT_CHAR_TEX_SIZE;
+        p++;
+    }
+    f += 16.0f * msgCtx->unk12098 * (D_801D0710[gSaveContext.language - 1] + 1);
+    p--;
+    *decodedBufPos = p;
+    *offset = o;
+    *arg3 = f;
+}
+
+void Message_LoadRupeesNES(GlobalContext* globalCtx, s16* decodedBufPos, s32* offset, f32* arg3, s16 singular) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    s16 p = *decodedBufPos;
+    s32 o = *offset;
+    f32 f = *arg3;
+
+    msgCtx->decodedBuffer[p] = 0x20;
+    p++;
+    Font_LoadCharNES(globalCtx, 0x52U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    msgCtx->decodedBuffer[p] = 0x52;
+    p++;
+    Font_LoadCharNES(globalCtx, 0x75U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    msgCtx->decodedBuffer[p] = 0x75;
+    p++;
+    Font_LoadCharNES(globalCtx, 0x70U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    msgCtx->decodedBuffer[p] = 0x70;
+    p++;
+    Font_LoadCharNES(globalCtx, 0x65U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    msgCtx->decodedBuffer[p] = 0x65;
+    p++;
+    Font_LoadCharNES(globalCtx, 0x65U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    msgCtx->decodedBuffer[p] = 0x65;
+    if (singular != 1) {
+        p++;
+        Font_LoadCharNES(globalCtx, 0x73U, o);
+        o += FONT_CHAR_TEX_SIZE;
+        msgCtx->decodedBuffer[p] = 0x73;
+        f += 16.0f * msgCtx->unk12098 * 6.0f;
+    } else {
+        f += 16.0f * msgCtx->unk12098 * 5.0f;
+    }
+    *decodedBufPos = p;
+    *offset = o;
+    *arg3 = f;
+}
+
+void Message_LoadTimeNES(GlobalContext* globalCtx, u8 arg1, s32* offset, f32* arg3, s16* decodedBufPos) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    s16 p = *decodedBufPos;
+    s32 o = *offset;
+    f32 f = *arg3;
+    u32 dayTime;
+    s16 digits[4];
+    f32 timeInSeconds;
+    s32 day;
+    s16 i;
+
+    if (arg1 == 0xCF) {
+        day = gSaveContext.day;
+        dayTime = 0x40000 - ((day % 5) << 16) - (u16)(-0x4000 + gSaveContext.time);
+    } else {
+        dayTime = 0x10000 - (u16)(-0x4000 + gSaveContext.time);
+    }
+    timeInSeconds = dayTime * (24.0f * 60.0f / 0x10000);
```

Comment:
```suggestion
    timeInSeconds = TIME_TO_MINUTES_F(dayTime);
```
This macro is currently not in your branch, but is in main. I also think `timeInSeconds` is actually `timeInMinutes`.